forch/faucet_state_collector.py

```diff
@@ -823,6 +823,23 @@ def process_port_learn(self, timestamp, name, port, mac, ip_addr):
                 devices_placement = DevicePlacement(switch=name, port=port, connected=True)
                 self._placement_callback(mac, devices_placement)
 
+    @_dump_states
+    # pylint: disable=too-many-arguments
+    def process_port_expire(self, timestamp, name, port, mac):
+        """process port learn event"""
+        with self.lock:
+            LOGGER.info('Learned entry %s at %s:%s expired.', mac, name, port)
+            if self.learned_macs:
+                entry = self.learned_macs.pop(mac, None)
+                if not entry:
+                    LOGGER.debug('Entry %s doesnt exist in learned macs dict', mac)
+            if name in self.switch_states:
```

Comment:
when would name never not be in switch_states?

Reply:
cleaned up.